test/cluster_client_slots_test.rb

```diff
@@ -26,8 +26,36 @@ def test_slot_class
     assert_nil slot.put(1, '127.0.0.1:7001')
   end
 
+  def test_slot_class_with_multiple_slot_ranges
+    slot = Redis::Cluster::Slot.new('127.0.0.1:7000' => [1..10, 30..40])
+
+    assert_equal false, slot.exists?(0)
+    assert_equal true, slot.exists?(1)
+    assert_equal true, slot.exists?(10)
+    assert_equal false, slot.exists?(11)
+    assert_equal true, slot.exists?(30)
+    assert_equal true, slot.exists?(40)
+    assert_equal false, slot.exists?(41)
+
+    assert_nil slot.find_node_key_of_master(0)
+    assert_nil slot.find_node_key_of_slave(0)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_master(1)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_slave(1)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_master(10)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_slave(10)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_slave(30)
+    assert_equal '127.0.0.1:7000', slot.find_node_key_of_slave(40)
+    assert_nil slot.find_node_key_of_master(11)
+    assert_nil slot.find_node_key_of_slave(11)
+    assert_nil slot.find_node_key_of_master(41)
+    assert_nil slot.find_node_key_of_slave(41)
+
+    assert_nil slot.put(1, '127.0.0.1:7001')
+    assert_nil slot.put(30, '127.0.0.1:7001')
+  end
```

Comment:
I'd say that a additional test case with `node_flags` and `with_replica` args is a nice-to-have. We want to make sure the scale read feature in multiple slot ranges case.

```ruby
Redis::Cluster::Slot.new({ '127.0.0.1:7000' => [1..10, 30..40], '127.0.0.1:7001' => [1..10, 30..40] },
                         { '127.0.0.1:7000' => 'master', '127.0.0.1:7001' => 'slave' }, true)
```